line-bot-servlet/src/main/java/com/linecorp/bot/servlet/LineBotCallbackRequestParser.java

```diff
@@ -58,19 +58,33 @@ public LineBotCallbackRequestParser(
     public CallbackRequest handle(HttpServletRequest req) throws LineBotCallbackException, IOException {
         // validate signature
         String signature = req.getHeader("X-Line-Signature");
+        final byte[] json = ByteStreams.toByteArray(req.getInputStream());
+        return handle(signature, new String(json, StandardCharsets.UTF_8));
+    }
+    
+    /**
+     * Parse request.
+     *
+     * @param signature X-Line-Signature header.
+     * @param payload Request body.
+     * @return Parsed result. If there's an error, this method sends response.
+     * @throws LineBotCallbackException There's an error around signature.
+     */
+    public CallbackRequest handle(String signature, String payload) throws LineBotCallbackException, IOException {
+        // validate signature
         if (signature == null || signature.length() == 0) {
             throw new LineBotCallbackException("Missing 'X-Line-Signature' header");
         }
-
-        final byte[] json = ByteStreams.toByteArray(req.getInputStream());
+        
         if (log.isDebugEnabled()) {
-            log.debug("got: {}", new String(json, StandardCharsets.UTF_8));
+            log.debug("got: {}", payload);
```

Comment:
Could you remove `if` block around this line?
There are no reason using `if` now.